### DISCUSSION TITLE
What is the default behavior for result calculation when no parameters are provided?

### DISCUSSION THREAD

**Discussion by jkaa2** [DaveSkender/Stock.Indicators#865] [Q&A] [answered]
Hi Dave,
could you please describe the default behavior for result calculation in case there is no look-back period parameter option, if one provide like 10000 quotes for an indicator, will the results cover all the quotes of just portion of the these quotes ?

**Comment by DaveSkender** [comment, marked as answer]
Yes, some indicators have default values set on parameters, when there is a standard value.  If you do not provide a parameter value, the default will be used.  You can see the default value in both 1) the documentation and 2) in the tooltip as you're typing in most IDEs.

![image](https://user-images.githubusercontent.com/8432125/184395638-670a9768-eaeb-470a-9878-6c10e378709c.png)
Example ref: [RSI docs](https://daveskender.github.io/Stock.Indicators/indicators/Rsi/#content)

The current v2 library returns all indicator values that can be calculated for the entire time series of `quotes` provided.  If you provide 1000 quotes, you'll get back 1000 result records for almost all indicators (there are a few exceptions).

With that said, any warmup periods that cannot be calculated will have `null` values.  Each indicator has a different amount of both incalculable warmup periods or potentially less precise data.  Our documentation will give you information on what to expect for each indicator.

Example for RSI:
![image](https://user-images.githubusercontent.com/8432125/184396619-93a9422d-d30f-4284-bd4c-3166d295c90b.png)

**Reply by jkaa2** [reply]
Thank you, I've posted the question because I provided like 8xxxx quotes for the zigzag indicator and got only 3xx entries as a result, seems like the quotes has no volatility.

**Reply by DaveSkender** [reply]
Sounds reasonable.  Defaults for ZigZag are 5% on Close price.  Quotes that trend for longer periods and don’t change direction very often wouldn’t produce as many high and low pivots.  For your example, you should get 8xxxx result records, but only 3xx show an `H` or `L` for `PointType`.

![](https://daveskender.github.io/Stock.Indicators/assets/charts/ZigZag.png)

If you’re using shorter timeframe bars, like hourly or minutely, you might want to consider a smaller `percentChange` value.
